pipeline/state_duration.go

```diff
@@ -0,0 +1,58 @@
+package pipeline
+
+import (
+	"time"
+
+	"github.com/influxdata/kapacitor/tick/ast"
+)
+
+// Compute the duration of a given state.
+// The state is defined via a lambda expression. For each consecutive point for
+// which the expression evaluates as true, the state duration will be
+// incremented by the duration between points. When a point evaluates as false,
+// the state duration is reset to 0.
+//
+// The state duration will be added as an additional field to each point. If the
+// expression evaluates as false, the field will not be added. If the expression
```

Comment:
The first point where the expression evaluates as true is 0. After all on the first point, there was no previous point, so it has been evaluating as true for 0s.
You might then argue we use a default of -1 instead, but even this can be incorrect as someone could do `order by time desc` and now the durations are negative instead of positive.
The only way to be absolutely certain the statement evaluated as false is to just not add the field at all.